src/components/Quiz.js

```diff
@@ -18,19 +18,56 @@ const Quiz = () => {
     setQuestions(quizData);
   }, [quizName]);
 
-  const handleAnswerSelect = e => {
-    const selectedAnswer = { [e.target.name]: Number(e.target.value) };
+  const handleCheckboxAnswers = (e, answers) => {
+    const questionIds = Object.keys(answers || {});
+    const hasQPreviouslySelectedAnswers =
+      questionIds && questionIds.includes(e.target.name);
+
+    let checkedAnswer;
+    if (hasQPreviouslySelectedAnswers) {
+      const previousSelectedAnswers = answers[e.target.name];
+      const isAnswerPreviouslySelected = previousSelectedAnswers.includes(
+        Number(e.target.value),
+      );
+      if (isAnswerPreviouslySelected) {
+        const removePreviousAnswer = previousSelectedAnswers.filter(
+          answer => answer !== Number(e.target.value),
+        );
+        checkedAnswer = {
+          [e.target.name]: removePreviousAnswer,
+        };
+      } else {
+        checkedAnswer = {
+          [e.target.name]: [...previousSelectedAnswers, Number(e.target.value)],
+        };
+      }
+    } else {
+      checkedAnswer = { [e.target.name]: [Number(e.target.value)] };
+    }
+    return checkedAnswer;
+  };
+
+  const handleAnswerSelect = (e, type) => {
+    let selectedAnswer;
+    if (Object.keys(userAnswers).length && type === 'checkbox') {
+      selectedAnswer = handleCheckboxAnswers(e, userAnswers);
+    } else {
+      selectedAnswer = { [e.target.name]: [Number(e.target.value)] };
+    }
     setUserAnswers({ ...userAnswers, ...selectedAnswer });
   };
 
   const checkUserAnswers = e => {
     e.preventDefault();
     const totalScore = questions.reduce((point, question) => {
-      const selectedAnswerId = userAnswers[question.id];
-      const selectedAnswerObj = question.answers.find(
-        answer => answer.id === selectedAnswerId,
+      const selectedAnswers = userAnswers[question.id];
+      const correctAnswers = question.answers
+        .map(answer => answer.isCorrect && answer.id)
+        .filter(Boolean);
+      const isRight = selectedAnswers.every(answer =>
+        correctAnswers.includes(answer),
```

Comment:
```suggestion
      const selectedAnswer = userAnswers[question.id];
      const isCorrect = question.answers.every(answer =>
        answer.isCorrect === selectedAnswer.includes(answer.id)
```
Good attempt at making the code more concise! Here is the code which checks if the answers are correct using just one statement. Also, I think renaming it from `isRight` to `isCorrect` is better since we're already using `isCorrect` as part of the quiz data, which is better for naming consistency, and `right` has a double meaning as the opposite of left.

Reply:
Done! Vicky was also showing me  a solution very close to this logic. Thank you both of you!